model_finetuning/PushIOProcessor.py

```diff
@@ -0,0 +1,265 @@
+				continue
+
+			if submission_id in known_ids:
+				logging.info(f"Skipping {submission_id} and detox for entry")
+				continue
+
+			submission['selftext'] = self.clean_text(submission.get('selftext'))
+
+			submissions_to_write.append(submission)
+
+			self.get_related_comments(submission_id)
+			continue
+
+		with db_instance.atomic():
+			models = [dict_to_model(Submission, submission, ignore_unknown=True) for submission in submissions_to_write]
+			jsons = [model_to_dict(model) for model in models]
+			Submission.insert_many(jsons).execute()
+
+		return
+
+	def process_comments(self, response: Response) -> None:
+
+		comments = self.get_data_from_response(response)
+
+		comment_ids = [comment.get('id') for comment in comments]
+
+		known_ids = [item.id for item in list(Comment.select(Comment.id).where(Comment.id.in_(comment_ids)))]
+
+		logging.info(f"{len(known_ids)} / {len(comment_ids)} comments found in Db for this set.")
+
+		if comments is None:
+			return
+
+		if len(comments) == 0:
+			return
+
+		comments_to_write = []
+
+		for comment in comments:
+
+			comment_id = comment.get('id')
+
+			if comment_id in known_ids:
+				logging.info(f"Skipping {comment_id} and detox for entry")
+				continue
+
+			comment['body'] = self.clean_text(comment.get('body'))
+
+			comment['is_url_only'] = self.set_is_url_only(comment)
+
+			if comment.get('body') in ['[removed]', '[deleted]']:
+				logging.debug(f"Comment Was removed or deleted for {comment_id}")
+				continue
+
+			if comment_id in known_ids:
+				logging.info(f"Skipping {comment_id} and detox for entry")
+				continue
+
+			comments_to_write.append(comment)
+
+		with db_instance.atomic():
+			# this is done to trick the serializer into accepting the fields. The correct thing to do would not direct
+			# intake the blob but that is a lot of work to do when this works well enough.
+			models = [dict_to_model(Comment, comment, ignore_unknown=True) for comment in comments_to_write]
+			jsons = [model_to_dict(model) for model in models]
+			Comment.insert_many(jsons).execute()
+
+		return
+
+	def run(self) -> None:
+		for start, end in self.loop_between_dates():
+			time_delta = (end.timestamp() - self.start_date_time.timestamp()) / self.calc_date_range()
+
+			logging.info(
+				f"downloading submission data from {start.date()} to {end.date()}... {round(time_delta * 100, 2)}%")
+
+			submission_search_link = (
+				'https://api.pushshift.io/reddit/submission/search/?subreddit={}&after={}&before={}&stickied=0&sort_type=score&sort=desc&limit={}&mod_removed=0&score{}')
+
+			submission_search_link = submission_search_link.format(self.subreddit, int(start.timestamp()),
+																   int(end.timestamp()), 100, self.score)
+
+			submission_response = self.get_with_retry(submission_search_link)
+
+			self.process_submission(submission_response)
+
+		return
+
+
+if __name__ == '__main__':
+	logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.DEBUG)
```

Comment:
We need to call `create_tables()` somewhere near the start of the main to create the tables if it's a brand new database.